milesightem310/sensor.go

```diff
@@ -144,3 +160,51 @@ func (n *em310Tilt) Close(ctx context.Context) error {
 func (n *em310Tilt) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	return n.node.Readings(ctx, extra)
 }
+
+// DoCommand implements the DoCommand for the em310Tilt.
+func (n *em310Tilt) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	testOnly := node.CheckTestKey(cmd)
+	if interval, intervalSet := cmd[intervalKey]; intervalSet {
+		if intervalFloat, floatOk := interval.(float64); floatOk {
+			return n.addIntervalToQueue(ctx, intervalFloat, testOnly)
+		}
+		return map[string]interface{}{}, fmt.Errorf("error parsing payload, expected float got %v", reflect.TypeOf(interval))
+	}
+	if _, ok := cmd[resetKey]; ok {
+		return n.addRestartToQueue(ctx, testOnly)
+	}
+
+	// do generic node if no sensor specific key was found
+	return n.node.DoCommand(ctx, cmd)
+}
+
+func (n *em310Tilt) addIntervalToQueue(ctx context.Context, interval float64, testOnly bool) (map[string]interface{}, error) {
+	// convert to the nearest second.
+	convertToSeconds := uint16(math.Round(interval * 60))
+
+	// create a byte array of size 2 and put the payload in as little endian.
+	bs := make([]byte, 2)
+	binary.LittleEndian.PutUint16(bs, convertToSeconds)
+
+	// ff byte is the channel, 03 is the message type byte for the downlink.
+	intervalString := "ff03"
+	for _, b := range bs {
+		intervalString = fmt.Sprintf("%s%02x", intervalString, b)
+	}
+
+	if testOnly {
+		return map[string]interface{}{intervalKey: intervalString}, nil
+	}
+
+	return n.node.SendDownlink(ctx, intervalString, false)
+}
```

Comment:
Is this also repeated code?

Reply:
nope, the implemetation for `addIntervalToQueue` is different for every model. the repeated parts are 

```
if testOnly {
		return map[string]interface{}{intervalKey: intervalString}, nil
	}

	return n.node.SendDownlink(ctx, intervalString, false)
```